### ISSUE TITLE
Feature request for page.waitForSelector() reverse

### ISSUE THREAD

**Issue by trevordmiller** [puppeteer/puppeteer#705]
I often want to wait for an element to disappear before continuing; right now, I'm adding selectors for both existing and removed situations (for example, when a modal is open and closed) so that I can use something like:

```javascript
await page.click('[data-testId="openModalButton"]')
await page.waitForSelector('[data-testId="modal"]')
await page.click('[data-testId="closeModalButton"]')
await page.waitForSelector('[data-testId="noModal"]')
```

But having to add a non-existent element (ie `noModal`) to check that a selector has been removed seems odd; could we get a `!` reverse/negate option or something like it? Like `page.waitForSelector(!{selector})`? So this could be:

```javascript
await page.click('[data-testId="openModalButton"]')
await page.waitForSelector('[data-testId="modal"]')
await page.click('[data-testId="closeModalButton"]')
await page.waitForSelector(!'[data-testId="modal"]')
```

Then adding empty DOM elements to verify a selector doesn't have a match wouldn't be necessary.

Thanks!

**Comment by JoelEinbinder** [comment]
Will this work for you?
```js
await page.waitFor(() => !document.querySelector(".cool-thing"));
```

**Comment by trevordmiller** [comment]
@JoelEinbinder Yeah that works but I think it would be helpful to have as a first-class option; maybe `page.waitForSelectorRemoval()`. I do this often and feel others would as well but maybe I'm in the minority here, in which feel free to close the issue.

**Comment by JoelEinbinder** [comment]
I'll close this issue, but if there is a lot of support for it we can revisit it.

**Comment by albertogasparin** [comment]
What about an option on `waitFor*` APIs? Something like:

```js
await page.waitForSelector('[data-testId="modal"]', { reverse: true })
```

(maybe with a better key name)

**Comment by ChristianDavis** [comment]
@albertogasparin I implemented a waitForSelector option for when an element is hidden, work for you? #967 

``` javascript
await page.waitForSelector('#selector', { hidden: true });
```

**Comment by Martin-Kuca** [comment]
@ChristianDavis I'm in the same need as @trevordmiller and I did try the { hidden: true } and unfortunately it doesn't work for me as I need the selector to disappear, not just be hidden. 

as you've written in #967 

> hidden: Waits for a selector to be present and hidden, opposite behavior of the option visible.

I think that the { reverse: true } option would be more appropriate

here is for example what the webdriverio's waitForExist reverse parameter does:

> reverse: if true it instead waits for the selector to not match any elements (default: false) 

I've previously used webdriverio for my tests and the reverse parameter has been essential for my tests. I am now using puppeteer and quite like it. However, the reverse parameter for waitForSelector would be really useful as it has, in my opinion, a very broad use case

**Comment by ChristianDavis** [comment]
@Martin-Kuca agreed, I implemented reverse in #967, trying to convince members it's useful

**Comment by Kaushikey1** [comment]
is this achievable  through current version
